### PR TITLE
Fix/builder duping 10

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Port to 1.10

Same as original pr

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/ColonyView.java [hunk 1]
```diff
@@ -64,7 +64,7 @@
      */
     private Set<Block> freeBlocks = new HashSet<>();
 
-    private double overallHappiness;
+    private double overallHappiness = 5;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/ColonyView.java]: MAJOR Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 92 issues
* ![MAJOR][MAJOR] 40 major
* ![MINOR][MINOR] 43 minor
* ![INFO][INFO] 9 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/3e0a3943b55b406606a0cc1751890d4dc5693f3d/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.231 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/3e0a3943b55b406606a0cc1751890d4dc5693f3d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/3e0a3943b55b406606a0cc1751890d4dc5693f3d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 82 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/3e0a3943b55b406606a0cc1751890d4dc5693f3d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Refactor this class so it has no more than 20 fields, rather than the 36 it currently has. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L641](https://github.com/Minecolonies/minecolonies/blob/3e0a3943b55b406606a0cc1751890d4dc5693f3d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L641): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1159](https://github.com/Minecolonies/minecolonies/blob/3e0a3943b55b406606a0cc1751890d4dc5693f3d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1159): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1160](https://github.com/Minecolonies/minecolonies/blob/3e0a3943b55b406606a0cc1751890d4dc5693f3d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1160): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1272](https://github.com/Minecolonies/minecolonies/blob/3e0a3943b55b406606a0cc1751890d4dc5693f3d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1272): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1274](https://github.com/Minecolonies/minecolonies/blob/3e0a3943b55b406606a0cc1751890d4dc5693f3d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1274): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1276](https://github.com/Minecolonies/minecolonies/blob/3e0a3943b55b406606a0cc1751890d4dc5693f3d/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1276): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 78 issues
* ![MAJOR][MAJOR] 32 major
* ![MINOR][MINOR] 40 minor
* ![INFO][INFO] 6 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/cca6bb74dab0512b5439b1b0cd2fa2a2cec26f53/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.231 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/cca6bb74dab0512b5439b1b0cd2fa2a2cec26f53/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 43 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/cca6bb74dab0512b5439b1b0cd2fa2a2cec26f53/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): class "Colony" has 82 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L43](https://github.com/Minecolonies/minecolonies/blob/cca6bb74dab0512b5439b1b0cd2fa2a2cec26f53/src/main/java/com/minecolonies/coremod/colony/Colony.java#L43): Refactor this class so it has no more than 20 fields, rather than the 36 it currently has. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1820)
1. ![MAJOR][MAJOR] [Colony.java#L641](https://github.com/Minecolonies/minecolonies/blob/cca6bb74dab0512b5439b1b0cd2fa2a2cec26f53/src/main/java/com/minecolonies/coremod/colony/Colony.java#L641): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L1159](https://github.com/Minecolonies/minecolonies/blob/cca6bb74dab0512b5439b1b0cd2fa2a2cec26f53/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1159): Assign this magic number 20 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1160](https://github.com/Minecolonies/minecolonies/blob/cca6bb74dab0512b5439b1b0cd2fa2a2cec26f53/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1160): Assign this magic number 60 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1272](https://github.com/Minecolonies/minecolonies/blob/cca6bb74dab0512b5439b1b0cd2fa2a2cec26f53/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1272): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1274](https://github.com/Minecolonies/minecolonies/blob/cca6bb74dab0512b5439b1b0cd2fa2a2cec26f53/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1274): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1276](https://github.com/Minecolonies/minecolonies/blob/cca6bb74dab0512b5439b1b0cd2fa2a2cec26f53/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1276): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marvin-bitterlich** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/988/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**